tests/unit/modules/network/a10/test_acos_config.py

```diff
@@ -71,30 +73,68 @@ def test_acos_config_lines(self):
         lines = ["ip dns primary 10.18.18.81"]
         set_module_args(dict(lines=lines))
         self.execute_module()
-        self.conn.edit_config.assert_called_with(
-            candidate=['ip dns primary 10.18.18.81'])
+        self.conn.get_diff = MagicMock(
+            return_value=self.cliconf_obj.get_diff(
+                candidate=lines, running=self.running_config, diff_match=self.match,
+                diff_ignore_lines=self.diff_ignore_lines
+            )
+        )
+        self.assertIn("ip dns primary 10.18.18.81",
+                      self.conn.get_diff.return_value['config_diff'])
+        self.assertTrue(self.conn.edit_config.called)
 
     def test_acos_config_multi_lines(self):
         lines = ["ip dns primary 10.18.18.81", "member rs1-test 80",
                  "slb server server2-test 5.5.5.11"]
         set_module_args(dict(lines=lines))
         self.execute_module()
-        self.conn.edit_config.assert_called_with(candidate=[
-            "ip dns primary 10.18.18.81", "member rs1-test 80", "slb server server2-test 5.5.5.11"])
+        self.conn.get_diff = MagicMock(
+            return_value=self.cliconf_obj.get_diff(
+                candidate=lines, running=self.running_config, diff_match=self.match,
+                diff_ignore_lines=self.diff_ignore_lines
+            )
+        )
+        self.assertIn("ip dns primary 10.18.18.81",
+                      self.conn.get_diff.return_value['config_diff'])
+        self.assertIn("member rs1-test 80",
+                      self.conn.get_diff.return_value['config_diff'])
+        self.assertIn("slb server server2-test 5.5.5.11",
+                      self.conn.get_diff.return_value['config_diff'])
+        self.assertTrue(self.conn.edit_config.called)
 
     def test_acos_config_before(self):
         lines = ["ip dns primary 10.18.18.19"]
         set_module_args(dict(lines=lines, before=["show avcs"]))
         self.execute_module()
-        self.conn.edit_config.assert_called_with(
-            candidate=['show avcs', 'ip dns primary 10.18.18.19'])
+        commands = ["show avcs", "ip dns primary 10.18.18.19"]
+        self.conn.get_diff = MagicMock(
```

Comment:
(Duplicate comment to make note of similar issue)
Why not just use the regular Mock? I don't see any magic methods being called upon.

Reply:
Yes, here Mock can be used. Replaced MagicMock with Mock now.
Resolved in PR#8